src/client/Microsoft.Identity.Client/Instance/GenericAuthority.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Microsoft.Identity.Client.Instance
+{
+    internal class GenericAuthority : Authority
+    {
+        internal const string OpenIdConfigurationEndpointSuffix = ".well-known/openid-configuration";
+        private static readonly ConcurrentDictionary<string, OpenIdConnectConfiguration> s_openIdConnectConfigurations = new();
+           
+        internal GenericAuthority(AuthorityInfo authorityInfo)
+            : base(authorityInfo)
+        {
+
+        }
+
+        internal override string TenantId => null;
+    
+        internal IDocumentRetriever DocumentRetriever { get; set; } = new HttpDocumentRetriever();
```

Comment:
`HttpDocumentRetriever` is not included.

Reply:
Alternatively, use `HttpDocumentRetrieve(HttpClient)` constructor, but I am hesitant to take a dependency on those libraries. Will get back to you on this one.

Reply:
Update on this - will discuss with folks on my side about using `HttpDocumentRetriever` dependency (with HttpClient). For now, let's continue to use it.

Reply:
@bgavrilMS Thanks I will. I just spotted a bug related to the fact that the Authority instance used to request the token is not the same one as the one used when using the builder, causing the DocumentRetriever property instance used being defaulted. Please advise about expected lifetimes of Authority instances. My understanding is that an AuthorityInfo instance should be able to new an Authority instance which implies that AuthorityInfo should store all required information to re build a new instance of a concrete Authority.

Reply:
`IDocumentRetriever` is no longer exposed in MSAL.NET public API